x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -75,6 +77,47 @@ export const signalRulesAlertType = ({
       } = params;
       // TODO: Remove this hard extraction of name once this is fixed: https://github.com/elastic/kibana/issues/50522
       const savedObject = await services.savedObjectsClient.get('alert', alertId);
+      const ruleStatusSavedObjects = await services.savedObjectsClient.find<
+        IRuleSavedAttributesSavedObjectAttributes
+      >({
+        type: ruleStatusSavedObjectType,
+        perPage: 6, // 0th element is current status, 1-5 is last 5 failures.
```

Comment:
Wouldn't 5 status be `[0,5] inclusive` where the first status starts at an index of 0, and the 5 ends with 4?

You have a mix of `perPage: 6` of 6 vs. 5 is this all correct?

Reply:
Yes I should have added a comment in the code here - I have to store a sixth historical item as a "current status". This is for the case where a status has failed five previous times and their current status is either executing or successful, I don't want to drop the 5th error to insert this executing / successful status. So I keep a 6th one (the most recent one) as a way to track the current status of the executing rule and the other 5 spots are for the 5 errors (if a rule has five errors in its history).

Reply:
I'm going to double check that I have my logic correct though.